### DISCUSSION TITLE
New Plugin: Disable Right Click

### DISCUSSION THREAD

**Discussion by gpsblues** [GetPublii/Publii#1773] [Useful Resources]
Hi everyone! I’ve just released a simple plugin called **[Disable Right Click](https://github.com/gpsblues/Publii-Plugin-Disable-right-click)**. 
This plugin disables right-click and text selection, which can be useful for discouraging image downloads, for example. It can be configured to work via CSS, JavaScript, or both.

This project is my first attempt at creating plugins, with plans to develop more complex ones in the future. It’s also a great excuse for me to learn my way around GitHub, a platform I’m still getting familiar with.

Thank you for your attention, and I hope some of you find it useful!

[Repository Disable Right Click](https://github.com/gpsblues/Publii-Plugin-Disable-right-click)

**Comment by bobmitro** [comment]
Excellent job @gpsblues 
We appreciate your contributions to our community - it will benefit many people. 
Thank you once again!

**Comment by kjmweb** [comment]
Nice plugin. But it doesn't work if the user disables the javascript. It should block the access to the page when the javascript is disabled and grant access when js is enabled again.

**Reply by gpsblues** [reply]
Hi!

Thank you very much for your feedback! 😊 This simple plugin is intended more as a basic deterrent rather than a real security system.

There is a more advanced solution on github, but I don't think it blocks the page, try searching for Plugin Publii.

**Comment by Vinc26** [comment]
Thanks so much! But there is a bug:

the code is `<link rel="stylesheet" href="${rendererInstance.siteConfig.domain}/media/plugins/disableRightClick/disableRC.min.css">`

But the real folder created on the server is `disable-right-click`

Sorry, I not used with Github: I don't know how to contribute by correct that by myself.

**Comment by gpsblues** [comment]
Hi,

Today I'm trying to verify, but actually I renamed the folder and probably didn't do it in the code.

Thanks a lot.

**Reply by gpsblues** [reply]
Hi,

v.1.0.1
Fixed a bug related to the incorrect name of the main folder.
https://github.com/gpsblues/Publii-Plugin-Disable-right-click/releases/tag/v.1.0.1

Due to the folder name change, it is recommended to uninstall previous versions and reinstall the plugin. This should prevent having duplicate plugins.

Thank you

https://github.com/gpsblues/Publii-Plugin-Disable-right-click/releases/tag/v.1.0.1

**Comment by Vinc26** [comment]
Ok thanks for 1.0.1 !

But now, I get this error in the safari consol:
`Did not parse stylesheet at 'https://aurelielamour.com/media/plugins/disable-right-click/disableRC.min.css' because non CSS MIME types are not allowed in strict mode.`

**Reply by Vinc26** [reply]
And as you cas see, it not the good path folder? 

Now it's the opposite of before: the code indicates `disable-right-click`, but the real folder is 'disableRightClick'.

**Reply by Vinc26** [reply]
Oh no. After disable plug, then publish, then enable plugin, then publish : ok now it works!!!

**Comment by EagerKoalaReader** [comment]
Just installed this - a God-send. We are very keen to stop our written work from being abused and this helps prevent that - delighted. Have only tested in preview mode so far though - will let you know if there are any issues in the fully live version.

**Comment by gpsblues** [comment]
Hi, 
keep in mind that this is a very basic protection. The purpose of the plugin is to discourage less experienced users, but as a security measure it's very easy to bypass.
